src/App.tsx

```diff
@@ -1,61 +1,110 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+
+const todosWithUsers = todosFromServer.map(todo => ({
+  ...todo,
+  user: usersFromServer.find(user => user.id === todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState(todosWithUsers);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState('0');
+  const [errors, setErrors] = useState<{ title: string; userId: string }>({
+    title: '',
+    userId: '',
+  });
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value.replace(/[^a-zA-Zа-яА-ЯіІїЇєЄ0-9\s]/g, ''));
+    if (errors.title) {
+      setErrors(prevErrors => ({ ...prevErrors, title: '' }));
+    }
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(event.target.value);
+    if (errors.userId) {
+      setErrors(prevErrors => ({ ...prevErrors, userId: '' }));
+    }
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const newErrors: { title: string; userId: string } = {
+      title: title ? '' : 'Please enter a title',
+      userId: userId !== '0' ? '' : 'Please choose a user',
+    };
+
+    if (newErrors.title || newErrors.userId) {
```

Comment:
```suggestion
    if (newErrors.title.trim() || newErrors.userId) {
```
to fix the bug with creating todo only with spaces